ros-planning/moveit2
# [Planning Pipeline Refactoring] #1 Simplify Adapter - Planner chain

## Description
Addresses #2408 
Requires https://github.com/ros-planning/moveit_msgs/pull/170

Related PRs:
- https://github.com/ros-planning/moveit_resources/pull/189
- https://github.com/ros-planning/moveit2_tutorials/pull/793
### How to test
Use https://github.com/ros-planning/moveit2_tutorials/pull/793 and https://github.com/ros-planning/moveit_resources/pull/189 and run 

```
ros2 launch moveit2_tutorials move_group.launch.py
```
and
```
ros2 launch moveit2_tutorials motion_planning_pipeline_tutorial.launch.py
```

Expected output
```bash
[motion_planning_pipeline_tutorial-1] [INFO] [1699454408.777050834] [motion_planning_pipeline_tutorial.remote_control]: Waiting to continue: Press 'next' in the RvizVisualToolsGui window to continue the demo
[motion_planning_pipeline_tutorial-1] [INFO] [1699454410.087785146] [motion_planning_pipeline_tutorial.remote_control]: ... continuing
[motion_planning_pipeline_tutorial-1] [INFO] [1699454410.087971200] [motion_planning_pipeline_tutorial]: Calling PlanningRequestAdapter 'ResolveConstraintFrames'
[motion_planning_pipeline_tutorial-1] [INFO] [1699454410.088272213] [motion_planning_pipeline_tutorial]: Calling PlanningRequestAdapter 'ValidateWorkspaceBounds'
[motion_planning_pipeline_tutorial-1] [WARN] [1699454410.088287442] [motion_planning_pipeline_tutorial.validate_workspace_bounds]: It looks like the planning volume was not specified. Using default values.
[motion_planning_pipeline_tutorial-1] [INFO] [1699454410.088442316] [motion_planning_pipeline_tutorial]: Calling PlanningRequestAdapter 'CheckStartStateBounds'
[motion_planning_pipeline_tutorial-1] [INFO] [1699454410.088596429] [motion_planning_pipeline_tutorial]: Calling PlanningRequestAdapter 'CheckStartStateCollision'
[motion_planning_pipeline_tutorial-1] [INFO] [1699454410.089114285] [moveit.ompl_planning.model_based_planning_context]: Planner configuration 'panda_arm[RRTConnectkConfigDefault]' will use planner 'geometric::RRTConnect'. Additional configuration parameters will be set when the planner is constructed.
[motion_planning_pipeline_tutorial-1] [INFO] [1699454410.124115820] [motion_planning_pipeline_tutorial]: Calling PlanningResponseAdapter 'AddTimeOptimalParameterization'
[motion_planning_pipeline_tutorial-1] [WARN] [1699454410.124140126] [moveit_trajectory_processing.time_optimal_trajectory_generation]: Invalid max_velocity_scaling_factor 0.000000 specified, defaulting to 1.000000 instead.
[motion_planning_pipeline_tutorial-1] [WARN] [1699454410.124149954] [moveit_trajectory_processing.time_optimal_trajectory_generation]: Invalid max_acceleration_scaling_factor 0.000000 specified, defaulting to 1.000000 instead.
[motion_planning_pipeline_tutorial-1] [INFO] [1699454410.125490595] [motion_planning_pipeline_tutorial]: Calling PlanningResponseAdapter 'ValidateSolution'
[motion_planning_pipeline_tutorial-1] [INFO] [1699454410.125781979] [motion_planning_pipeline_tutorial]: Calling PlanningResponseAdapter 'DisplayMotionPath'
[motion_planning_pipeline_tutorial-1] [INFO] [1699454410.125976980] [motion_planning_pipeline]: Visualizing the trajectory
```
### Changes
- Simplify pipeline component call chain to a sequence of two vectors + a planner
- Disallow adapters to call the planner
-  Remove FixStartStatePathConstraints :arrow_right: The problem of connecting a start state to the start of a constrained motion should be solved with MTC today 
- Add `PipelineState` publisher which publishes the intermediate state between the pipeline stages. 
  - To test it, set the `progress_topic` some topic name you'd like to see the intermediate the intermediate results published to before the planning pipeline instances are constructed. The parameter is evaluated during the construction of a planning pipeline instance
 - Make the moveit error code the sole source to report success or failure
   - `moveit::core::MoveItStatus PlanningRequestAdapter::adapt(..` Plan request adapters return a status containing error information
   - PlannerContext `virtual void solve(MotionPlanResponse& res) = 0;` uses the response to communicate success or failure
   - PlanRequestAdapter `virtual void adapt(..., planning_interface::MotionPlanResponse& res)` also uses the response for communication
   

### TODOs
- [ ] Create humble release + branch for moveit_resources
- [x] Update setup assistant
- [x] Add instructions for migration guide

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"
